### DISCUSSION TITLE
Write access to a single parameter

### DISCUSSION THREAD

**Discussion by Herbie911** [fredlcore/BSB-LAN#617] [Q&A] [answered]
Hi,

I tried to enable the write access to one or two parameter like it is describe in the manual ( replacing DEFAULT_FLAG by 0 (Zero) in the parameter line ).
But it doesn't work.
In contrast to the manual I found the parameter lines at the end of BSB_LAN_custom_defs.h and not in the BSB_LAN_defs.h but I think that doesn't matter.
What works is the replacing the code ' #define DEFAULT_FLAG FL_SW_CTL_RONLY ' by ' #define DEFAULT_FLAG 0 ' to enable write access to all parameter. But that is not what I want.

I work with 
- Arduino Due
- BSB Interface
- BSB-Lan Version 3.3.2

Has anyone an idea what the reason can be ?

**Comment by fredlcore** [comment]
Please post what exactly you have changed in which file, otherwise there's nothing I can do.

**Reply by Herbie911** [reply]
In the file BSB_LAN_custom_defs.h I made this change:
![19 02](https://github.com/fredlcore/BSB-LAN/assets/160403367/582cb875-0350-4aed-aa90-d7487dc00d5c)

**Comment by fredlcore** [comment, marked as answer]
You didn't post your BSB_LAN_config.h file, so I assume there are no changes there. If you have DEFAULT_FLAG set to FL_SW_CTL_RONLY there, that means that you also have to enable writing in the configuration of the webinterface. If you enabled that, it will enable writing most or all parameters. What you want to achieve is done by setting DEFAULT_FLAG to FL_RONLY and then make the changes you made above.

**Reply by Herbie911** [reply]
now it's working perfect .
I missunderstood this comment line : '// use #define DEFAULT_FLAG FL_RONLY to run the program always in read-only mode.' in the BSB_LAN_config.h file.

Thanks for quickly support

**Reply by fredlcore** [reply]
Since all parameters in BSB_LAN_custom_defs.h are either FL_RONLY or DEFAULT_FLAG, the above statement is correct. If, however, you change that file, then obviously this statement is no longer correct, but that is the exception, not the rule...
